src/OmniSharp.Script/ScriptHelper.cs

```diff
@@ -43,26 +43,65 @@ public class ScriptHelper
         private static readonly CSharpParseOptions ParseOptions = new CSharpParseOptions(LanguageVersion.Latest, DocumentationMode.Parse, SourceCodeKind.Script);
 
         private readonly Lazy<CSharpCompilationOptions> _compilationOptions;
-        private readonly MetadataReferenceResolver _resolver = ScriptMetadataResolver.Default;
+        private readonly Lazy<CSharpCommandLineArguments> _commandLineArgs;
+        private readonly ScriptOptions _scriptOptions;
+        private readonly IOmniSharpEnvironment _env;
+        private readonly ILogger _logger;
+        private readonly bool _isDesktopClr;
 
-        public ScriptHelper(IConfiguration configuration = null)
+        public ScriptHelper(ScriptOptions scriptOptions, IOmniSharpEnvironment env, ILoggerFactory loggerFactory, bool isDesktopClr)
         {
-            _configuration = configuration;
+            _scriptOptions = scriptOptions ?? throw new ArgumentNullException(nameof(scriptOptions));
+            _env = env ?? throw new ArgumentNullException(nameof(env));
+
+            _logger = loggerFactory.CreateLogger<ScriptProjectSystem>();
             _compilationOptions = new Lazy<CSharpCompilationOptions>(CreateCompilationOptions);
-            InjectXMLDocumentationProviderIntoRuntimeMetadataReferenceResolver();
+            _commandLineArgs = new Lazy<CSharpCommandLineArguments>(CreateCommandLineArguments);
+            _isDesktopClr = isDesktopClr;
+        }
+
+        private CSharpCommandLineArguments CreateCommandLineArguments()
+        {
+            var scriptRunnerParserProperty = typeof(CSharpCommandLineParser).GetProperty("ScriptRunner", BindingFlags.Static | BindingFlags.NonPublic);
```

Comment:
FWIW, I don't think you need this anymore. `CSharpCommandLineParser.Script` appears to be public in Roslyn 2.7.0.